gentle_gnomes/src/azavea.py

```diff
@@ -0,0 +1,86 @@
+from typing import Any, Dict, Iterator, List, NamedTuple, Optional, Union
+
+import requests
+
+BASE_URL = 'https://app.climate.azavea.com/api'
+
+
+class City(NamedTuple):
+    name: str
+    admin: str
+    id: int
+
+    def __str__(self):
+        return f'{self.name}, {self.admin}'
+
+
+class Client:
+    """Client for interacting with the Azavea Climate API."""
+
+    def __init__(self, token: str):
+        self.session = requests.Session()
+        self.session.headers = {'Authorization': f'Token {token}'}
+
+    def _get(self, endpoint: str, **kwargs) -> Union[Dict, List]:
+        response = self.session.get(BASE_URL + endpoint, ** kwargs)
+        response.raise_for_status()
+
+        return response.json()
+
+    @staticmethod
+    def _set_param(name: str, value: Any, params: Optional[Dict]) -> Dict:
+        """Set a parameter if it isn't already set."""
+        value_dict = {name: value}
+
+        if not params:
+            params = value_dict
+        elif name not in params:
+            params.update(value_dict)
+
+        return params
+
+    def get_cities(self, **kwargs) -> Iterator[City]:
+        """Return all available cities."""
+        params = kwargs.get('params')
+        params = self._set_param('page', 1, params)
```

Comment:
Good naming conventions! It seems kinda funky write a method for that.

Something like this could work:

    params = kwargs.get('params', {})
    params = {**{'page': 1}, **params}  # `page` param defaults to 1